Dockerfile

```diff
@@ -0,0 +1,53 @@
+ARG PYTHON_VERSION=3.7
+ARG ALPINE_VERSION=3.13
+
+FROM python:${PYTHON_VERSION}-alpine${ALPINE_VERSION}
+
+WORKDIR /app
+VOLUME [ "/data" ]
+
+ENV GRAB_SITE_INTERFACE=0.0.0.0
+ENV GRAB_SITE_PORT=29000
+ENV GRAB_SITE_HOST=127.0.0.1
+EXPOSE 29000
+
+RUN apk add --no-cache \
+		git \
+		gcc \
+		libxml2-dev \
+		musl-dev \
+		libxslt-dev \
+		g++ \
+		re2-dev \
+		libffi-dev \
+		openssl-dev \
+		patch \
+		cargo \
+	&& ln -s /usr/include/libxml2/libxml /usr/include/libxml \
+	&& addgroup -S grab-site \
+	&& adduser -S -G grab-site grab-site \
+	&& chown -R grab-site:grab-site $(pwd) \
+	&& mkdir -p /data \
+	&& chown -R grab-site:grab-site /data
+
+USER grab-site:grab-site
+ENV PATH="/app:$PATH"
+ENTRYPOINT [ "entrypoint.sh" ]
+CMD [ "gs-server" ]
+
+# TODO: resolve dependencies before loading library code to take advantage of build caching
+# 	setup.py requires libgrabsite/__init__.py (__version__ property) to work
+
+COPY --chown=grab-site:grab-site . .
+
+RUN pip install . \
+	&& chmod +x entrypoint.sh
```

Comment:
`COPY` preserves the executable bit, so I'd just make `entrypoint.sh` executable in the repo.